flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/changelog/StateChangelogClient.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.changelog;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.state.KeyGroupRange;
+import org.apache.flink.runtime.state.StateChangelogHandle;
+
+/**
+ * {@link StateChangelogWriter} factory. Scoped to a single entity (e.g. a SubTask or
+ * OperatorCoordinator).
+ */
+@Internal
+public interface StateChangelogClient<Handle extends StateChangelogHandle<?>>
```

Comment:
Why is factory class named client? I had no idea what `StateChangelogClient` is doing looking at the name (I was more expecting it to be an internal class used by the changelog to perform the writes), and it's a bit funny that javadoc makes it easy to understand in the first sentence by calling it a `factory` :)

Reply:
Thanks for pointing out. This is a leftover from the previous design which assumed a dedicated component.
I'll rename it to `StateChangelogWriterFactory`.